avaframe/out3Plot/plotUtils.py

```diff
@@ -862,7 +878,23 @@ def fetchContourCoords(xGrid, yGrid, data, level):
     contourP = plt.contour(xGrid, yGrid, data, levels=[level])
 
     contourDictXY = {}
-    for i in range(len(contourP.allsegs[0])):
-        contourDictXY['line%s_%d' %(level, i)] = {'x': contourP.allsegs[0][i][:, 0], 'y':contourP.allsegs[0][i][:, 1]}
+    # loop over all segments of the contour line of the required level
+    # check matplotlib version to fetch coordinates of line
+    mVersionStr = (matplotlib.__version__).split('.')[0:2]
+    mVersion = float(mVersionStr[0] + mVersionStr[1])
+    if mVersion < 3.8:
+        for i in range(len(contourP.allsegs[0])):
+            contourDictXY['line%s_%d' %(level, i)] = {'x': contourP.allsegs[0][i][:, 0], 'y':contourP.allsegs[0][i][:, 1]}
+    else:
+        # use codes of path
+        labelID = 0
+        for ind, val in enumerate(contourP.get_paths()[0].codes):
+            if val == 1:
+                labelID = labelID + 1
+                contourDictXY['line%s_%d' % (level, labelID)] = {'x': [contourP.get_paths()[0].vertices[ind, 0]],
+                                                                 'y': [contourP.get_paths()[0].vertices[ind, 1]]}
+            else:
+                contourDictXY['line%s_%d' % (level, labelID)]['x'].append(contourP.get_paths()[0].vertices[ind, 0])
+                contourDictXY['line%s_%d' % (level, labelID)]['y'].append(contourP.get_paths()[0].vertices[ind, 1])
 
```

Comment:
Why is this necessary? Is labelID always either 0 or 1? and line 894 and 897 do basically the same?! Or am I missing something?

Reply:
multple lines, val ==1 comes at each start of level

Reply:
because now all the vertices are provided as one array